staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -259,6 +260,11 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			if compiled.FieldPathError != nil {
+				klog.V(2).ErrorS(compiled.FieldPathError, "fieldPath check failed")
+			} else if len(rule.FieldPath) > 0 {
+				fldPath = compiled.NormalizedRuleFieldPath.SetRoot(fldPath)
```

Comment:
I'm finding the SetRoot usage here really hard to reason about... is there a reason this isn't just:

```go
if len(compiled.NormalizedRuleFieldPath) > 0 {
  fldPath = fldPath.Child(compiledNormalizedRuleFieldPath)
}
```